src/v/coproc/event_listener.cc

```diff
@@ -196,11 +202,40 @@ ss::future<> event_listener::do_start() {
         } else {
             _active_ids.clear();
             _offset = model::offset(0);
+            co_await advertise_state_update();
         }
     }
+
     co_await do_ingest();
 }
 
+ss::future<> event_listener::boostrap_status_topic() {
+    /// Bootstrap the coproc status topic. At mininum the wasm engine must
+    /// be up for this part of code to be reached.
+    kafka::creatable_topic copro_status_topic{
+      .name = model::coprocessor_status_topic,
+      .num_partitions = 1,
+      .replication_factor = 3,
+      .configs{
+        kafka::createable_topic_config{
+          .name = "cleanup.policy", .value = "compact"},
+        kafka::createable_topic_config{
+          .name = "compression.type", .value = "zstd"}}};
+    auto responses = co_await _client.create_topic(
+      std::move(copro_status_topic));
```

Comment:
are there any exceptional futures that need to be handled here?

Reply:
I checked, afaik Ben's client catches the exceptions and converts them into the appropriate error code within the responses array. @BenPope is that correct?

Reply:
I don't always have an error decoder in the chain. Check what schema registry does in ensure_started. I think it explicitly handles the error type to check if the topic already exists.

Reply:
what about this one https://github.com/vectorizedio/redpanda/blob/dev/src/v/kafka/client/client.cc#L262